core/unit_test/TestLocalDeepCopy.hpp

```diff
@@ -1042,4 +1042,76 @@ TEST(TEST_CATEGORY, local_deepcopy_rangepolicy_layoutright) {
   }
 }
 #endif
+
+namespace Impl {
+template <typename T, typename SHMEMTYPE>
+using ShMemView =
+    Kokkos::View<T, Kokkos::LayoutRight, SHMEMTYPE, Kokkos::MemoryUnmanaged>;
+
+struct DeepCopyScratchFunctor {
+  DeepCopyScratchFunctor(
+      Kokkos::View<double*, TEST_EXECSPACE::memory_space> check_view_1,
+      Kokkos::View<double*, TEST_EXECSPACE::memory_space> check_view_2)
+      : check_view_1_(check_view_1),
+        check_view_2_(check_view_2),
+        N_(check_view_1.extent(0)) {}
+
+  KOKKOS_INLINE_FUNCTION void operator()(
+      Kokkos::TeamPolicy<TEST_EXECSPACE,
+                         Kokkos::Schedule<Kokkos::Dynamic>>::member_type team)
+      const {
+    using ShmemType = TEST_EXECSPACE::scratch_memory_space;
+    auto shview =
+        Impl::ShMemView<double**, ShmemType>(team.team_scratch(1), N_, 1);
+
+    Kokkos::parallel_for(
+        Kokkos::TeamThreadRange(team, N_), KOKKOS_LAMBDA(const size_t& index) {
+          auto thread_shview = Kokkos::subview(shview, index, Kokkos::ALL());
+          Kokkos::Experimental::local_deep_copy(thread_shview, index);
+        });
+    TEST_EXECSPACE().fence();
```

Comment:
that is the wrong fence. That shouldn't even compile ...
call `team.team_barrier()`

Reply:
Looking at https://github.com/kokkos/kokkos/blob/80fcf8560f5adcace1bc4944b7e1135c9df89d8d/core/src/Kokkos_CopyViews.hpp#L1873-L1890 the appropriate fences are already in place.

Reply:
oh right because you now went to the team based once. I think the one which doesn't take a team handle is not fencing or?

Reply:
Yes, the ones without team handle are not fencing.